code/modules/atmospherics/machinery/portable/pipe_scrubber.dm

```diff
@@ -0,0 +1,159 @@
+/obj/machinery/portable_atmospherics/pipe_scrubber
+	name = "pipe scrubber"
+	icon_state = "pipe_scrubber"
+	density = TRUE
+	max_integrity = 250
+	volume = 200
+	///The internal air tank obj of the mech
+	var/obj/machinery/portable_atmospherics/canister/internal_tank
+	///Is the machine on?
+	var/on = FALSE
+	///What direction is the machine pumping to (into scrubber or out to the port)?
+	var/direction = PUMP_IN
+	///the rate the machine will scrub air
+	var/volume_rate = 1000
+	///List of gases that can be scrubbed
+	var/list/scrubbing = list(
+		/datum/gas/plasma,
+		/datum/gas/carbon_dioxide,
+		/datum/gas/nitrous_oxide,
+		/datum/gas/bz,
+		/datum/gas/nitrium,
+		/datum/gas/tritium,
+		/datum/gas/hypernoblium,
+		/datum/gas/water_vapor,
+		/datum/gas/freon,
+		/datum/gas/hydrogen,
+		/datum/gas/healium,
+		/datum/gas/proto_nitrate,
+		/datum/gas/zauker,
+		/datum/gas/halon,
+	)
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/Initialize(mapload)
+	internal_tank = new(src)
+	. = ..()
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/Destroy()
+	var/turf/my_turf = get_turf(src)
+	my_turf.assume_air(air_contents)
+	my_turf.assume_air(internal_tank.air_contents)
+	SSair.stop_processing_machine(internal_tank)
+	qdel(internal_tank)
+	return ..()
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/return_analyzable_air()
+	return list(
+		air_contents,
+		internal_tank.air_contents
+	)
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/disconnect()
+	on = FALSE
+	return ..()
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/click_alt(mob/living/user)
+	return FALSE
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/attackby(obj/item/item, mob/user, params)
+	return TRUE
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/update_icon_state()
+	icon_state = on ? "[initial(icon_state)]_active" : initial(icon_state)
+	return ..()
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/process_atmos()
+	if(take_atmos_damage())
+		excited = TRUE
+		return ..()
+	if(!on)
+		return ..()
+	if(isnull(connected_port))
+		return ..()
+	if(direction == PUMP_IN)
+		scrub(air_contents)
+	else
+		internal_tank.air_contents.pump_gas_to(air_contents, PUMP_MAX_PRESSURE)
```

Comment:
shouldn't this make it excited or am I insane

Reply:
but we do need to keep processing? like that's what excited is for

Reply:
It's in `ui_act()`.
```
			if(on)
				SSair.start_processing_machine(src)
				SSair.start_processing_machine(internal_tank)
```

Reply:
no like we start processing but if excited is not set to true and nothing else is doing work we will stop processing

Reply:
I see, so I need to set `excited = TRUE` every atmos frame if I want the atom to process the next frame too... 

I have added it before the direction check, but it was working well even without it. Maybe missing something.